core/commands/pin/remotepin.go

```diff
@@ -167,13 +167,19 @@ NOTE: a comma-separated notation is supported in CLI for convenience:
 		}
 
 		// Prepare Pin.origins
-		// Add own multiaddrs to the 'origins' array, so Pinning Service can
-		// use that as a hint and connect back to us (if possible)
+		// If CID in blockstore, add own multiaddrs to the 'origins' array
+		// so pinning service can use that as a hint and connect back to us.
 		node, err := cmdenv.GetNode(env)
 		if err != nil {
 			return err
 		}
-		if node.PeerHost != nil {
+
+		isInBlockstore, err := node.Blockstore.Has(rp.Cid())
```

Comment:
@lidel this seems fine. I don't think we're introducing any new problems on the CoreAPI/Node boundary, but wanted to double check. WDYT?

The boundary issues we generally run into are that IIRC accessing the node directly doesn't respect the `--offline` flag, however I think we may already have that issue given that we use `node.PeerHost` below.